src/Providers/Models/DTO/ModelMetadata.php

```diff
@@ -0,0 +1,208 @@
+     *
+     * @since n.e.x.t
+     *
+     * @return string The model ID.
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Gets the model's display name.
+     *
+     * @since n.e.x.t
+     *
+     * @return string The model name.
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * Gets the model's supported capabilities.
+     *
+     * @since n.e.x.t
+     *
+     * @return CapabilityEnum[] The supported capabilities.
+     */
+    public function getSupportedCapabilities(): array
+    {
+        return $this->supportedCapabilities;
+    }
+
+    /**
+     * Gets the model's supported configuration options.
+     *
+     * @since n.e.x.t
+     *
+     * @return SupportedOption[] The supported options.
+     */
+    public function getSupportedOptions(): array
+    {
+        return $this->supportedOptions;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public static function getJsonSchema(): array
+    {
+        return [
+            'type' => 'object',
+            'properties' => [
+                self::KEY_ID => [
+                    'type' => 'string',
+                    'description' => 'The model\'s unique identifier.',
+                ],
+                self::KEY_NAME => [
+                    'type' => 'string',
+                    'description' => 'The model\'s display name.',
+                ],
+                self::KEY_SUPPORTED_CAPABILITIES => [
+                    'type' => 'array',
+                    'items' => [
+                        'type' => 'string',
+                        'enum' => CapabilityEnum::getValues(),
+                    ],
+                    'description' => 'The model\'s supported capabilities.',
+                ],
+                self::KEY_SUPPORTED_OPTIONS => [
+                    'type' => 'array',
+                    'items' => SupportedOption::getJsonSchema(),
+                    'description' => 'The model\'s supported configuration options.',
+                ],
+            ],
+            'required' => [self::KEY_ID, self::KEY_NAME, self::KEY_SUPPORTED_CAPABILITIES, self::KEY_SUPPORTED_OPTIONS],
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     *
+     * @return ModelMetadataArrayShape
+     */
+    public function toArray(): array
+    {
+        return [
+            self::KEY_ID => $this->id,
+            self::KEY_NAME => $this->name,
+            self::KEY_SUPPORTED_CAPABILITIES => array_values(array_map(
+                static fn(CapabilityEnum $capability): string => $capability->value,
+                $this->supportedCapabilities
+            )),
+            self::KEY_SUPPORTED_OPTIONS => array_values(array_map(
+                static fn(SupportedOption $option): array => $option->toArray(),
+                $this->supportedOptions
+            )),
```

Comment:
Why do these use `array_values()`? The two properties are documented to be indexed arrays anyway.

Reply:
Are you sure those are the only two places in the codebase that are supposed be indexed arrays but we don't validate them? I guess my point is, for now let's just go with the simple approach: no validation and expect indexed arrays to be indexed arrays.

I think polyfills for things like that would be great (also `str_contains` etc.), but let's add those via a separate PR purely focused on setting them up. We could also combine that with adding validation for any input arrays that _should_ be arrays.

Reply:
I'm good with that. In any case we'll remove these and handle the validation elsewhere. 👍